.github/workflows/main.yml

```diff
@@ -1,4 +1,9 @@
 name: CI
+# https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#concurrency
+# https://docs.github.com/en/developers/webhooks-and-events/events/github-event-types#pullrequestevent
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number }}
```

Comment:
You can add more `-${{ ... }}` to this for example with tags if you want, e.g. [`${{ github.event.type }}`](https://docs.github.com/en/developers/webhooks-and-events/events/github-event-types#event-object-common-properties). That would probably be safer in case you push a tag and then quickly push a commit to `main` the main one won't kill the tag one (since push to main should have `push` type and tag should have `create` type, I think?).

Reply:
Yeah that seems reasonable.